spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -97,6 +97,8 @@ protected Catalog buildIcebergCatalog(String name, CaseInsensitiveStringMap opti
     Configuration conf = SparkSession.active().sessionState().newHadoopConf();
     Map<String, String> optionsMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
     optionsMap.putAll(options);
+    optionsMap.put("app-id", SparkSession.active().sparkContext().applicationId());
+    optionsMap.put("user", SparkSession.active().sparkContext().sparkUser());
```

Comment:
Should these keys be defined in `CatalogProperties`?

Reply:
I was hesitant to move them to a central place initially, but we'll probably will need those at least for Flink as well, so let's move them to `CatalogProperties`